### DISCUSSION TITLE
Stationarity assumption

### DISCUSSION THREAD

**Discussion by jjakenichol** [jakobrunge/tigramite#144] [Q&A] [answered]
In the assumptions tutorial it lists stationarity. Is that statistical stationarity, e.g. summary statistics of each time series do not change in time, or is that causal structure stationarity, e.g. the underlying causal structure does not change?

If not the former, does a time series need any transforming/preprocessing, like a standard scaler?

Many ML algorithms need stationarity or a standard scale, so I originally assumed that was the case. I recently found a talk you all gave where you mentioned causal structure stationarity so I thought I'd come for clarity.

Thank you!

Edit: In looking over Glymour et al.'s Review of Causal Discovery Methods Based on Graphical Models, I found this "...the system may be non-stationary (i.e., the probability distributions of variables conditional on their causes may change, and even the causal relations may change)..." on page 9. That implies to me that we're talking about both stationarity of the underlying causal structure and the timeseries' distributions in the stationarity assumption. Is it ever a mistake to apply any transforms to the input timeseries? Can PCMCI, or other causal network learning algorithms, handle trend and seasonality in the data?

**Comment by jakobrunge** [comment, marked as answer]
It's a good question!

For practical purposes, depending on the assumptions of the conditional independence tests, PCMCI assumes full stationarity of the distribution since only then is, for example, partial correlation well calibrated.  You may build your own independence test that doesn't require this, of course!

And in theory, for infinite sample sizes, we only require "causal [structure] stationarity" as it is defined in the Chaos paper from 2018 since only the conditional independencies have to prevail, not the particular dependencies. But this whole issue is not theoretically fully solved yet, I would say.

If there are trends or seasonality, I would think of these as unobserved confounders that you have to treat explicitly. See the Chaos paper for an example. Then you would remove the trend/seasonality just as you would condition on a confounder.

Note that transformations such as differencing your data change the meaning of the causal variables and will change the graph.

**Reply by jjakenichol** [reply]
Thank you, I hadn't found the Chaos paper yet. All of that makes a lot of sense. I was differencing several timeseries, but not all (some were already stationary), and I think that was a mistake. To make a consistent interpretation between links, the variables should all represent the same type of data: differenced data are deltas from timestep to timestep. That definitely changes the meaning of the causal relationships. Removing trend/seasonality as a way to remove unobserved confounders is a great framing - I think that would be an intuitive explanation for why many ML models require it and I've never read it explained that way.
